src/internal/snapshot/server_internal_test.go

```diff
@@ -0,0 +1,65 @@
+package snapshot
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/snapshot"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestCreateSnapshot(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	c := NewTestClient(t)
+	metadata := map[string]string{"key": "value"}
+	resp, err := c.CreateSnapshot(ctx, &snapshot.CreateSnapshotRequest{
+		Metadata: metadata,
+	})
+	if err != nil {
+		t.Fatalf("create snapshot RPC: %v", err)
+	}
+	t.Log(resp)
+}
+
+func TestListSnapshot(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	c := NewTestClient(t)
+	createSnapshots(t, ctx, c)
+	time.Sleep(2 * time.Second)
```

Comment:
I don't think we need to sleep this long.  The timestamps are nanosecond-precision.  A few milliseconds at most will ensure the records have different timestamps.

I would avoid even that sleep.  List all the rows, then find the second most recent one, and ask traversal to start there, and make sure allRows[1:] == rowsStartingAtTheSecondOne.